lib/smooth/include/smooth/core/util/DmaFixedBuffer.h

```diff
@@ -28,26 +22,29 @@ limitations under the License.
 
 namespace smooth::core::util
 {
-    template<typename T, size_t Size>
+    template<typename T, size_t Size, uint32_t MallocCapType>
```

Comment:
The way templates work, having the memory capabilities as a template parameter results in multiple types if different capabilities are used. This in turn results in large code size, which is a scarce resource on embedded system. I suggest passing this as a constructor argument instead.